### ISSUE TITLE
Feature: Add support for multiple regex engines

### ISSUE THREAD

**Issue by MikeLemo1** [tomiis4/Hypersonic.nvim#17]
Useful for anyone who works across multiple platforms, not an expert in regex for now for suffered from the struggle 
here's a list of some of the available engines: https://www.regexmagic.com/flavors.html 

The ones I used the most are: Vim, Lua, Python, JS, PHP and maybe in the future Golang regex engine.

Does the plugin already support nvim's lua regex btw?

**Comment by tomiis4** [comment]
Hello, I'm sorry for the late response, it fully support LUA and PHP but other engines may work too.

**Comment by MikeLemo1** [comment]
Yeah but escaping chars and wildcard expansion inside sets doesn't work it is different than the regex treatment you have in say regex101.com  So it's not exactly the same
